lib/ingress/permission_rule.rb

```diff
@@ -35,8 +35,14 @@ def subject_matches?(given_subject)
         "*" == subject
     end
 
+    def subject_class?(subject)
+      [Class, Module].include? subject.class
+    end
+
     def conditions_match?(user, given_subject, options)
```

Comment:
What I would probably do is not even call the `conditions_match?` method if we don't care about conditions e.g.

```
    def match?(given_action, given_subject, user, options = {})
      return false unless action_matches?(given_action)
      return false unless subject_matches?(given_subject)

      conditions_match?(user, given_subject, options) unless ignore_coditions?(given_subject)
    end

    private

    def ignore_coditions?(given_subject)
      subject_class?(given_subject) && subject_class?(@subject)
    end
```